Added zbar recipe, first revision.

barctor opened the pull request:
Adds a recipe for [zbar](http://zbar.sourceforge.net/) providing:
* Main package: cmd:zbarimg - command line program to scan QR codes & barcodes in image files
* Devel package: lib:libzbar - C library/headers exposing API to scan barcodes & QR codes

Currently builds package without GTK, QT, or Python support.

extrowerk commented:
Not tested, but looks good.

Begasus commented:
Latest release is 0.23.90 according to repology: https://repology.org/project/zbar/versions
Hosted at: https://github.com/mchehab/zbar/

barctor commented:
Thanks @Begasus I didn't know development had moved to github.
I've amended the commit to use the latest release from the linked git repo.
As far as tests go:
* Main package: zbarimg is able to scan images and print any QR code data found
* Devel package: i can build a project that references zbar.h but have yet to test the actual API to see if it works as intended

Begasus commented on media-gfx/zbar/zbar-0.23.90.recipe:
```diff
@@ -0,0 +1,110 @@
+ARCHITECTURES="all !x86_gcc2 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="0.3.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
```
libVersion comes after commandSuffix: see https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering

Begasus commented on media-gfx/zbar/zbar-0.23.90.recipe:
```diff
@@ -0,0 +1,110 @@
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libMagick++_7.Q16HDRI$secondaryArchSuffix
+	lib:libMagickCore_7.Q16HDRI$secondaryArchSuffix
+	lib:libMagickWand_7.Q16HDRI$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
```
Sort alphabetically (also in BUILD_REQUIRES)

Begasus commented on media-gfx/zbar/zbar-0.23.90.recipe:
```diff
@@ -0,0 +1,110 @@
+	"
+
+BUILD_REQUIRES="
+	imagemagick${secondaryArchSuffix}_devel
+	devel:libMagick++_7.Q16HDRI$secondaryArchSuffix
+	devel:libMagickCore_7.Q16HDRI$secondaryArchSuffix
+	devel:libMagickWand_7.Q16HDRI$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	haiku${secondaryArchSuffix}_devel
```
Changes:
```diff
- 	imagemagick${secondaryArchSuffix}_devel (not needed)
+	haiku${secondaryArchSuffix}_devel
	devel:libMagick++_7.Q16HDRI$secondaryArchSuffix
	devel:libMagickCore_7.Q16HDRI$secondaryArchSuffix
	devel:libMagickWand_7.Q16HDRI$secondaryArchSuffix
	devel:libiconv$secondaryArchSuffix
	devel:libjpeg$secondaryArchSuffix
	devel:libintl$secondaryArchSuffix
-	haiku${secondaryArchSuffix}_devel
```

Begasus commented on media-gfx/zbar/zbar-0.23.90.recipe:
```diff
@@ -0,0 +1,110 @@
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtool
```
Change for secondary architecture:
```diff
- 	cmd:libtool
+ 	cmd:libtoolize$secondaryArchSuffix
```

Begasus commented on media-gfx/zbar/zbar-0.23.90.recipe:
```diff
@@ -0,0 +1,110 @@
+BUILD()
+{
+	autoreconf -vfi
+	LDFLAGS="-lintl" runConfigure ./configure \
```
We need to move the binary to $prefix/bin here too:
```diff
- 	LDFLAGS="-lintl" runConfigure ./configure \
+ 	LDFLAGS="-lintl" runConfigure --omit-dirs binDir ./configure \
+ 	 	--bindir=$commandBinDir \

barctor commented:
Thanks for the review @Begasus 
I think I have addressed all the issues you found in the recipe.
I'll also make sure I read through the guide more thoroughly before submitting another recipe :)

Begasus commented on media-gfx/zbar/zbar-0.23.90.recipe:
```diff
@@ -0,0 +1,111 @@
+	zbar${secondaryArchSuffix}_devel = $portVersion
+	devel:libzbar$secondaryArchSuffix = $libVersionCompat
+	"
+
```
No new line between PROVIDES_devel and REQUIRES_devel
```diff
-
```

barctor commented:
Sorted the PROVIDES_devel and REQUIRES_devel spacing, and removed the extra space between libVersion and PROVIDES

Begasus commented:
Thanks! :+1:
